bids/variables/entities.py

```diff
@@ -33,6 +40,9 @@ def add_variable(self, variable):
         variable : BIDSVariable
             The Variable to add to the list.
         """
+        for e, val in self.base_ents.items():
```

Comment:
Can't `self.base_ents` be `None` (L32)? If not, why not?

Reply:
Yep. Arguably all of this logic / this check is not necessary but I thought it would help prevent issues like this in the future.